sqlglot/dialects/duckdb.py

```diff
@@ -1207,6 +1209,33 @@ def objectinsert_sql(self, expression: exp.ObjectInsert) -> str:
 
             return self.func("STRUCT_INSERT", this, kv_sql)
 
+        def _prepare_startswith_arg(self, arg: exp.Expression) -> None:
+            """Prepare argument for STARTS_WITH by converting to VARCHAR.
+
+            ByteString literals are converted to regular string literals to avoid
+            BLOB casting by the generator. Non-VARCHAR types are cast to VARCHAR.
+            """
+            # Annotate types if needed for type-based casting
+            if not arg.type:
+                from sqlglot.optimizer.annotate_types import annotate_types
+
+                annotate_types(arg, dialect=self.dialect)
+
+            # Convert ByteString to String literal before generation
+            # ByteStrings get typed as UNKNOWN and would be wrapped in CAST(...AS BLOB) by generator
+            if isinstance(arg, exp.ByteString):
+                arg.replace(exp.Literal.string(arg.this))
+            # Cast non-VARCHAR types to VARCHAR
```

Comment:
I don't think this logic is necessary?

1. We don't have to annotate types here, since transpiling this successfully from bigquery to duckdb requires that we run type inference first, and hence we should have the types available. For duckdb -> duckdb, we don't need types– the varchar/unknown branch should suffice since the type will always be unknown and we won't cast, hence preserving the sql @ roundtrip
2. Doesn't look like we should deal with bytestring as a special case. Despite a bit verbose, doing a double-cast is probably the safer way to go.